app/api/puzzle/submit/route.ts

```diff
@@ -6,16 +6,30 @@ import RatingCalculator from '@/src/rating/RatingCalculator';
 import { Puzzle } from '@/types/lichess-api';
 import { NextRequest } from 'next/server';
 import { RatingColl } from '../../../../models/RatingColl';
-import { RoundColl } from '@/models/RoundColl';
 import {
   getDefaultRating,
   getPuzzleRating,
   getThemeRatings,
 } from '@/src/rating/getRating';
-import { AllRoundColl } from '@/models/AllRoundColl';
 import { UserThemeColl } from '@/models/UserThemeColl';
 import addRound from './addRound';
 import { RatingHistory } from '@/models/RatingHistory';
+import { ActivePuzzleColl } from '@/models/ActivePuzzle';
+import { updateLeitner } from '@/src/LeitnerIntance';
+
+const REVIEW_SCALING_FACTOR = 0.7;
+
+// If the puzzle was a review, we scale the rating change by 70%.
+// We apply this both to user's overrall rating and to theme ratings.
+// TODO: This also reduces rating decrease for failed reviews (the idea being
+// not to penalise a user being bad at something), but one can argue that
+// we should be harsher instead on a failed review.
+// TODO: Should we scale the rating deviation and volatility as well? Not
+// sure how safe it is to solely scale the rating.
+const scaleRatingDelta = (oldRating: number, newGlicko: Rating): void => {
+  const delta = newGlicko.rating - oldRating;
+  newGlicko.rating = oldRating + REVIEW_SCALING_FACTOR * delta;
+};
```

Comment:
cc: @yichenchong @ahmedmahmud since this is kinda a big decision that'd affect theme ratings - does this sound OK? i think it makes sense

Reply:
I like the idea of scaling, but I do agree if the rating decrease should be scale up if a review if failed.

Reply:
Thanks, I'll mark this as an issue for later for the sake of time